pkg/client/apiutil/apimachinery.go

```diff
@@ -146,11 +152,24 @@ func GVKForObject(obj runtime.Object, scheme *runtime.Scheme) (schema.GroupVersi
 	return gvks[0], nil
 }
 
+// DefaultHTTPClient is a helper function that sets the HTTP client based on a rest config if it is not already set.
+func DefaultHTTPClient(config *rest.Config, httpClient *http.Client) (*http.Client, error) {
+	if httpClient != nil {
+		return httpClient, nil
+	}
```

Comment:
That's probably fine, we have "defaulting" functions in lots of places around the codebase. The main thing I would like to avoid is to pass `nil` parameters around